### PR TITLE
Install ipfs as a normal extension

### PR DESCRIPTION
**Pull request by yrliou**
Fix https://github.com/brave/brave-browser/issues/3950
Fix https://github.com/brave/brave-browser/issues/4218

Instead of installing as a component extension, this PR installs ipfs companion using a webstore installer so manifest could be localized and the extension option page could be shown normally under brave://extensions.

Toggle on the switch would prompt to install IPFS companion extension if it was't installed before. If already installed, it will enable the extension.
Toggle off the switch would disable IPFS companion extension (but not uninstall it). To fully remove (uninstall) the IPFS companion extension after it is installed, remove the extension through brave://extensions.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Test Plan:
1. Visit brave://settings and toggle IPFS companion switch, a prompt as below will be shown if IPFS companion extension is not currently installed.
Screen Shot 2019-04-24 at 8 18 15 AM

2. Click "Cancel" in the prompt, the IPFS companion switch should be changed back to off.

3. Toggle it on again, and click "Add extension" in the prompt to install it, IPFS companion extension should now be installed and enabled.

4. Right click IPFS companion switch icon to open options page, it should shown as
Screen Shot 2019-04-24 at 8 25 58 AM

5. Toggle it off again and visit brave://extensions, IPFS companion extension should be shown disabled.

6. Enable the IPFS companion switch toggle in brave://settings again, IPFS companion extension should be shown enabled in brave://extensions.

7. Remove IPFS companion extension in brave://extensions, IPFS companion extension switch toggle in brave://settings should be shown off.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/ui/webui/settings/brave_default_extensions_handler.cc [hunk 1]
```diff
@@ -83,6 +85,15 @@ void BraveDefaultExtensionsHandler::SetHangoutsEnabled(
   }
 }
 
+bool BraveDefaultExtensionsHandler::IsExtensionInstalled(
+    const std::string extension_id) {
```
Comment (emerick) [inline browser/ui/webui/settings/brave_default_extensions_handler.cc]: I think you probably want const std::string& here?

File: browser/ui/webui/settings/brave_default_extensions_handler.h [hunk 2]
```diff
@@ -25,6 +25,8 @@ class BraveDefaultExtensionsHandler : public settings::SettingsPageUIHandler {
   void SetHangoutsEnabled(const base::ListValue* args);
   void SetIPFSCompanionEnabled(const base::ListValue* args);
 
+  bool IsExtensionInstalled(const std::string extension_id);
```
Comment (emerick) [inline browser/ui/webui/settings/brave_default_extensions_handler.h]: Can it be a const method?

**Comment by GeetaSarvadnya** [comment]
@yrliou  Step 5:  Toggle IPFS off again - do you mean off the IPFS from below popup? If so, after making IPFS off, IPFS companion extension is NOT disabled in brave://extensions. Can you please clarify the point 5 in the above test plan

image

**Comment by yrliou** [comment]
@GeetaSarvadnya No, it is not the option page but the switch we introduced in brave://settings.

**Pull request merged**